staging/src/k8s.io/mount-utils/mount_linux_test.go

```diff
@@ -545,3 +549,220 @@ func mountArgsContainOption(t *testing.T, mountArgs []string, option string) boo
+					return "", err
+				}
+
+				symlink := filepath.Join(tmpDir, "symlink")
+				err = os.Symlink(f.Name(), symlink)
+				if err != nil {
+					return "", err
+				}
+				return symlink, nil
+			},
+		},
+		"path is a mount point": {
+			expectedErr:        nil,
+			expectedMountPoint: true,
+			prepare: func(t *testing.T) (string, error) {
+				// /proc is a known mount-point and since mount-utils heavily relies on /proc/mounts
+				// it is a good option where we can test for mount points in non-privileged mode.
+				return "/proc", nil
+			},
+		},
+		"path is not a mount point": {
+			expectedErr:        nil,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				source, err := os.MkdirTemp("", "source")
+				if err != nil {
+					return "", err
+				}
+				t.Cleanup(func() {
+					os.RemoveAll(source)
+				})
+				return source, nil
+			},
+		},
+		"path is a magiclink": {
+			expectedErr:        unix.ELOOP,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				// only /proc/1/exe is valid
+				return "/proc/1/exe", nil
+			},
+		},
+		"path does not exist": {
+			expectedErr:        os.ErrNotExist,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				f, err := os.CreateTemp(os.TempDir(), "unexisting")
+				if err != nil {
+					return "", err
+				}
+				err = os.Remove(f.Name())
+				if err != nil {
+					return "", err
+				}
+				return f.Name(), nil
+			},
+		},
+		"parent directory does not exist": {
+			expectedErr:        os.ErrNotExist,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				return "/random-path/which/does-not/exist", nil
+			},
+		},
+	}
+
+	hasOpenAt2Support := isOpenAt2Supported()
+	if !hasOpenAt2Support {
+		t.Logf("openat2 is not supported, will use mock function openAt2Mock")
+	} else {
+		t.Log("openat2 is supported, will use openat2 syscall.")
+	}
+	for name, test := range tests {
+		path, err := test.prepare(t)
+		if err != nil {
+			t.Errorf("test (%s) failed during prepare with error: %s", name, err.Error())
+			continue
+		}
+
+		// let's not use mock if openat2 is supported on a CI.
+		var isMountPoint bool
+		if hasOpenAt2Support {
+			isMountPoint, err = isMountPointUsingOpenAt2(path, nil)
+		} else {
+			isMountPoint, err = isMountPointUsingOpenAt2(path, openAt2Mock)
+		}
+
+		if isMountPoint != test.expectedMountPoint {
+			t.Errorf("test (%s) has a mis-match for expected mount point (%t) and got (%t)",
+				name, test.expectedMountPoint, isMountPoint)
+		}
+
+		if test.expectedErr != nil {
+			if err == nil || !errors.Is(err, test.expectedErr) {
+				t.Errorf("test (%s) has a mis-match for expected error (%s) and got (%s)", name,
+					test.expectedErr, err)
+			}
+		} else if err != nil {
+			t.Errorf("test (%s) did not expect any error to be returned but got (%s)", name,
+				err)
+		}
+	}
+}
+
+func openAt2Mock(dirfd int, path string, how *unix.OpenHow) (int, error) {
```

Comment:
I question the utility of using dependency injection for testing of a syscall, but ok... :)